File: frontend/src/objects/auth/AuthSession.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type { SessionToken } from '@/objects/core/SharedObjects'
import type { AuthAccount } from '@/objects/auth/AuthAccount'

export type AuthSession = {
  token: SessionToken
  user: AuthAccount
}

File: backend/src/main/scala/services/auth/objects/AuthSession.scala
package services.auth.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import system.objects.*

final case class AuthSession(token: SessionToken, user: AuthAccount)
object AuthSession:
  given Encoder[AuthSession] = deriveEncoder
  given Decoder[AuthSession] = deriveDecoder
